### PR TITLE
[Merged by Bors] - feat: Support podOverrides

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

*Please add a description here. This will become the commit message of the merge request later.*

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Comment by sbernauer** [comment]
Started https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/opa-operator-it-custom/69/

File: rust/crd/src/lib.rs [hunk 1]
```diff
@@ -29,6 +29,15 @@ pub const CONFIG_FILE: &str = "config.yaml";
 
 #[derive(Snafu, Debug)]
 pub enum Error {
+    #[snafu(display("the role {role} is not defined"))]
+    CannotRetrieveOpaRole { role: String },
```
Comment (adwk67) [inline rust/crd/src/lib.rs:33]: This does not appear to be used.
```suggestion
```

File: rust/crd/src/lib.rs [hunk 2]
```diff
@@ -237,6 +246,30 @@ pub enum OpaRole {
 }
 
 impl OpaCluster {
+    /// Returns a reference to the role. Raises an error if the role is not defined.
```
Comment (adwk67) [inline rust/crd/src/lib.rs:249]:
```suggestion
    /// Returns a reference to the role.
```

**Review by adwk67 — APPROVED** [review]
LGTM! Just the removal of dead code.

**Comment by sbernauer** [comment]
bors r+
Thx! Tests passed

**Pull request closed without merging**